### PR TITLE
Add  test for /sum/body and implementation and tests for /roll endpoint

### PR DESCRIPTION
**Pull request by nhanaa**

### CODE REVIEW

File: hello-spring/src/test/java/org/folio/hello/tests/RollTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      .get(getRequestUrl("/roll"));
+    response.then().statusCode(200);
+
+    Integer rollResult = Integer.parseInt(response.getBody().asString());
+    assertThat(rollResult, is(both(greaterThan(0)).and(lessThan(7))));
+  }
+
+  @Test
+  public void testRollPost() {
+    Random random = new Random();
```
Comment (ncovercash) [inline hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: SonarLint should've given you a warning in your IDE:
image

We generally don't like to use random values in tests, since then randomness can cause test pass or failure which is a recipe for painful debugging.

I'd recommend just using a constant value (maybe one or two different ones in different methods) to test this endpoint.

File: hello-spring/src/test/java/org/folio/hello/tests/RollTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import java.util.Random;
+import java.util.Arrays;
+
+import io.restassured.response.Response;
+
+import org.folio.hello.BaseIntegrationTest;
+import org.junit.jupiter.api.Test;
+
+public class RollTest extends BaseIntegrationTest {
```
Comment (ncovercash) [inline hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: Regular classes (and methods) don't need `public` (or any access modifiers); sonarlint should have warned about this.

The only exception is for when test classes are extended (like BaseIntegrationTest); in this case, it has to be marked public

File: hello-spring/src/test/java/org/folio/hello/tests/RollTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    Response response = ra()
+      .header("Content-Type", "application/json")
+      .body(numRoll.toString())
+      .post(getRequestUrl("/roll"));
+    response.then().statusCode(200);
+
+    Integer[] rollResultsArray = response.getBody().as(Integer[].class);
+    Iterable<Integer> rollResults = Arrays.asList(rollResultsArray);
+    assertThat(rollResults, everyItem(is(both(greaterThan(0)).and(lessThan(7)))));
```
Comment (ncovercash) [inline hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: Love this! Great use of hamcrest

File: hello-spring/src/main/java/org/folio/hello/RollController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public ResponseEntity<Integer> rollGet() {
+        Integer response = random.nextInt(6) + 1;
+
+        return new ResponseEntity<>(response, HttpStatus.OK);
+    }
+
+    /** {@inheritDoc}} */
+    @Override
+    public ResponseEntity<List<Integer>> rollPost(Integer n) {
+        List<Integer> response = new ArrayList<Integer>(n);
```
Comment (ncovercash) [inline hello-spring/src/main/java/org/folio/hello/RollController.java]: Very minor nitpick (but also one sonar should've caught), you can use `new ArrayList<>(n)` on the right side, since the compiler knows from the left side that you're making a `List<Integer>`.

**Review by ncovercash — CHANGES_REQUESTED** [review]
Looks great overall! A couple of nitpicks but really good overall

File: hello-spring/src/test/java/org/folio/hello/tests/RollTest.java (hunk 1 above)
Comment (nhanaa) [reply hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: Huh for some reasons sonarlint stopped working in my VSCode (it was working before so I never noticed). I'll take a look and fix accordingly.

Comment (ncovercash) [reply hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: Interesting - does it say anything in the problems tab? And did you copy in the sonarlint rules list from our vscode config?

Comment (nhanaa) [reply hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: It doesn't say anything. I'll take a look at the config file when I get back from dinner. Maybe I missed something.

Comment (nhanaa) [reply hello-spring/src/test/java/org/folio/hello/tests/RollTest.java]: Oh yeah I got it now. Since I'm using a new computer, I haven't copied the rules list on the current computer yet. It is working now.

**Review by ncovercash — CHANGES_REQUESTED** [review]
Looks good!  For completeness, mind adding test(s) for `POST /roll` that does some other cases (such as 0 and a large number, like 100?)

**Comment by ncovercash** [comment]
tests are pedantic, but we really want to make sure we cover every edge and corner case! i've seen cases before like this where someone hardcoded `5` in the api endpoint response code, then used the same number in the test, so it seemed fine but the parameter wasn't working at all.  the more tests the better!

**Comment by nhanaa** [comment]
I'll get on it tomorrow.

**Comment by nhanaa** [comment]
I tried adding a test for negative input, but not sure if that is the right way to do it.

**Comment by ncovercash** [comment]
> I tried adding a test for negative input, but not sure if that is the right way to do it.

i mean it looks like it works, i guess the default error from spring is a 400 (`ArrayList` throws when you construct it with a negative argument).  Looks good though, perfectly reasonable outcome!

Have you gotten formatting setup with VS Code?  It seems like the formatter is not working (four-space indentation, among other things)

**Comment by nhanaa** [comment]
I completely forgot about changing the setting after I switched over to my new computer. Sorry about that. It should all be good now.
